### PR TITLE
Refactor API key store interface

### PR DESCRIPTION
**Pull request by nicholaspcr**
#### Summary

Reference #5587

#### Changes

- Separate DeleteAPIKey into its own method

#### Testing

Unit tests

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [ ] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Review by adriansmares — APPROVED** [review]
Same curiosity here - was the API not added just to keep this small ?

File: pkg/identityserver/bunstore/api_key_store.go [hunk 1]
```diff
@@ -355,6 +341,45 @@ func (s *apiKeyStore) UpdateAPIKey(
 	return updatedPB, nil
 }
 
+func (s *apiKeyStore) DeleteAPIKey(
+	ctx context.Context, entityID *ttnpb.EntityIdentifiers, pb *ttnpb.APIKey,
+) error {
+	ctx, span := tracer.StartFromContext(ctx, "DeleteAPIKeys", trace.WithAttributes(
```
Comment (adriansmares) [inline pkg/identityserver/bunstore/api_key_store.go]: `DeleteAPIKey`.

**Comment by nicholaspcr** [comment]
>Same curiosity here - was the API not added just to keep this small ?

Partly that and trying to keep myself to what was on the issue, you already saw that one of the branches had a bit of the rpc implementation. I'm just trying to have things a bit more separated, wondering if this is easier to review or not

**Comment by adriansmares** [comment]
> 'm just trying to have things a bit more separated, wondering if this is easier to review or not

It's fine to break it down into smaller PRs that you may chain or do in another order.

**Pull request merged**